src/relay.cpp

```diff
@@ -800,10 +815,11 @@ void server_callback(evutil_socket_t fd, short event, void *arg) {
     sockaddr_in6 from;
     socklen_t len = sizeof(from);
     int32_t data = 0;
-    static uint8_t message_buffer[4096];
+    static uint8_t message_buffer[BUFFER_SIZE];
 
-    if ((data = recvfrom(config->local_sock, message_buffer, 4096, 0, (sockaddr *)&from, &len)) == -1) {
-        syslog(LOG_WARNING, "recv: Failed to receive data from server\n");
+    if ((data = recvfrom(config->local_sock, message_buffer, BUFFER_SIZE, 0, (sockaddr *)&from, &len)) == -1) {
+        syslog(LOG_ERR, "recv: Failed to receive data from server\n");
+        return;
     }
 
     auto msg = parse_dhcpv6_hdr(message_buffer);
```

Comment:
auto msg = parse_dhcpv6_hdr(message_buffer);
counters[msg->msg_type]++;

We need add protection for the range of msg->msg_type, in case of malformed dhcpv6 packets.
Same as the place in function server_callback_dual_tor( )